src/components/content/deployedServices/myServices/query/useListRegisteredServiceQuery.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { DeployedService, ServiceVendorService } from '../../../../../xpanse-api/generated';
+
+export default function useListRegisteredServicesQuery(
+    category: DeployedService.category,
+    csp: DeployedService.csp,
+    serviceName: string,
+    version: string,
+    serviceHostingType: DeployedService.serviceHostingType
+) {
+    return useQuery({
+        queryKey: ['listDeployedServices', category, csp, serviceName, version, serviceHostingType],
+        queryFn: () =>
+            ServiceVendorService.listServiceTemplates(category, csp, serviceName, version, serviceHostingType),
```

Comment:
there is a problem here. the `listServiceTemplates` can be accessed with users with "ISV" role only. But currently we are using it in the "end user" context. It works for you because your user has all roles. 
We must here use a service that is available for end user. 

What we must do is to extend the `DeployedService` and `DeployedServiceDetails` to have the ID of the service template that was used to order.